x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java

```diff
@@ -77,71 +77,82 @@ protected void masterOperation(Task task,
+                    .getTrainedModel()
+                    .targetType()) == false) {
+                listener.onFailure(ExceptionsHelper.badRequestException(
+                    "Model [{}] inference config type [{}] does not support definition target type [{}]",
+                    config.getModelId(),
+                    config.getInferenceConfig().getName(),
+                    config.getModelDefinition().getTrainedModel().targetType()));
+                return;
+            }
+
+            Version minCompatibilityVersion = config
                 .getModelDefinition()
                 .getTrainedModel()
-                .targetType()) == false) {
-            listener.onFailure(ExceptionsHelper.badRequestException(
-                "Model [{}] inference config type [{}] does not support definition target type [{}]",
-                config.getModelId(),
-                config.getInferenceConfig().getName(),
-                config.getModelDefinition().getTrainedModel().targetType()));
-            return;
+                .getMinimalCompatibilityVersion();
+            if (state.nodes().getMinNodeVersion().before(minCompatibilityVersion)) {
+                listener.onFailure(ExceptionsHelper.badRequestException(
+                    "Definition for [{}] requires that all nodes are at least version [{}]",
+                    config.getModelId(),
+                    minCompatibilityVersion.toString()));
+                return;
+            }
         }
 
-        Version minCompatibilityVersion = config
-            .getModelDefinition()
-            .getTrainedModel()
-            .getMinimalCompatibilityVersion();
-        if (state.nodes().getMinNodeVersion().before(minCompatibilityVersion)) {
-            listener.onFailure(ExceptionsHelper.badRequestException(
-                "Definition for [{}] requires that all nodes are at least version [{}]",
-                config.getModelId(),
-                minCompatibilityVersion.toString()));
-            return;
-        }
 
-        TrainedModelConfig trainedModelConfig = new TrainedModelConfig.Builder(config)
+
+
+        TrainedModelConfig.Builder trainedModelConfig = new TrainedModelConfig.Builder(config)
             .setVersion(Version.CURRENT)
             .setCreateTime(Instant.now())
             .setCreatedBy("api_user")
-            .setLicenseLevel(License.OperationMode.PLATINUM.description())
-            .setEstimatedHeapMemory(config.getModelDefinition().ramBytesUsed())
-            .setEstimatedOperations(config.getModelDefinition().getTrainedModel().estimatedNumOperations())
-            .build();
+            .setLicenseLevel(License.OperationMode.PLATINUM.description());
+        if (hasModelDefinition) {
+            trainedModelConfig.setEstimatedHeapMemory(config.getModelDefinition().ramBytesUsed())
+                .setEstimatedOperations(config.getModelDefinition().getTrainedModel().estimatedNumOperations());
+        }
```

Comment:
🤔 

we will have to rethink this. It would be good to somehow indicate the work and size required for a pytorch model.

Reply:
++ we will need a measure of the model size in TrainedModelConfig to allocate the persistent task by memory